### PR TITLE
Add missing `resetMesh()` on non-primary rank for `precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput`

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
## Main changes of this PR

Adds missing API calls in the following tests:

```
precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput
precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput2LI
```

## Motivation and additional information

I became aware of this inconsistency due to a failing test in #2220. The changes implemented here result in passing tests in #2220.

## Author's checklist

* [ ] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [ ] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [ ] I added a test to cover the proposed changes in our test suite.
* [ ] For breaking changes: I documented the changes in the appropriate [porting guide](https://precice.org/couple-your-code-porting-overview.html).
* [ ] I stuck to C++17 features.
* [ ] I stuck to CMake version 3.22.1.
* [ ] I squashed / am about to squash all commits that should be seen as one.
* [ ] I ran the systemtests by adding the label `trigger-system-tests` (may be skipped if minor change)

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
@fsimonis is it allowed to call `resetMesh()` only on some ranks of the participant? If not: This should raise an error in the first place.

**Comment by fsimonis** [comment]
This should work without the remesh. I guess when using substeps, we end up with rank 0 containing no initial time sample due to the reset and rank 1 containing an initial time sample.

**Comment by BenjaminRodenberg** [comment]
Ok. So how can we then fix this issue? Is the original test valid (bug in the implementation) or is the original test invalid (we need a safeguard to make sure all ranks calls `resetMesh()`).

I'm not very deep in the whole remeshing topic but I think it would be a good idea to force all ranks to call `resetMesh()` at the same time. Otherwise we get a whole set of additional edge cases that we have to explicitly test for (like the one we were obviously missing here).

What's the motivation for one rank calling `resetMesh()` while others don't? I assume there is some overhead when calling `resetMesh()` but is this really worth the additional complexity and risking bugs?

File: tests/remeshing/parallel-explicit/helper.hpp [hunk 1]
```diff
@@ -215,6 +215,8 @@ inline void runResetInput(testing::TestContext &context)
           .initialize()
           .write({1.01, 1.02})
           .advance()
+          .resetMesh()
+          .setVertices({2.0, y, 3.0, y})
```
Comment (BenjaminRodenberg) [inline tests/remeshing/parallel-explicit/helper.hpp:219]: To also make the goal of this PR a bit clearer: Was it intended in the original design of this test that the non-primary ranks of `"A"` do not call `resetMesh` or was it an oversight?

**Comment by fsimonis** [comment]
To clarify, it is a feature that only a subset of ranks need to call `resetMesh()`.

My best guess is that using substeps and remeshing, we currently end up with different time samples. This needs some special handling.

**Comment by BenjaminRodenberg** [comment]
I did some tests of this branch in combination with https://github.com/precice/precice/tree/fix-remesh-with-substeps (potential fix). I ran the problematic tests that I updated in #2225 via `ctest -R precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput
Test project /home/benjamin/Programming/precice/build`

After merging 45300c353f460e5fbfbdd2f5e9fd08aed9defc14 into this branch:

```
Test project /home/benjamin/Programming/precice/build
    Start 775: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput
1/3 Test #775: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput .........   Passed    3.27 sec
    Start 776: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput2LI
2/3 Test #776: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput2LI ......   Passed    0.48 sec
    Start 777: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInputSerial
3/3 Test #777: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInputSerial ...   Passed    0.46 sec
```

After reverting 078f7cffb175535216be11921b918083c9f3f1e4, i.e. resetting the tests to the original state while using the potential fix 45300c353f460e5fbfbdd2f5e9fd08aed9defc14:

```
Test project /home/benjamin/Programming/precice/build
    Start 775: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput
1/3 Test #775: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput .........   Passed    3.20 sec
    Start 776: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput2LI
2/3 Test #776: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput2LI ......   Passed    0.47 sec
    Start 777: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInputSerial
3/3 Test #777: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInputSerial ...   Passed    0.48 sec
```

If I also revert the fix 45300c3:

```
Test project /home/benjamin/Programming/precice/build
    Start 775: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput
1/3 Test #775: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput .........***Failed    5.18 sec
    Start 776: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput2LI
2/3 Test #776: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput2LI ......***Failed    2.52 sec
    Start 777: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInputSerial
3/3 Test #777: precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInputSerial ...   Passed    0.48 sec

33% tests passed, 2 tests failed out of 3

Label Time Summary:
integration    =   8.18 sec*proc (3 tests)
remeshing      =   8.18 sec*proc (3 tests)

Total Test time (real) =   8.23 sec

The following tests FAILED:
	775 - precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput (Failed)
	776 - precice.Integration/Remeshing/ParallelExplicit/ChangedMapping/RemeshInput2LI (Failed)
```

## Summary

I think the test meets the expectation and solves the original problem. Thanks @fsimonis :+1:

**Comment by BenjaminRodenberg** [comment]
This change of the test is not needed anymore due to the fix 45300c35. See https://github.com/precice/precice/pull/2220#issuecomment-2737444400 for details. Closing this PR.
